sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/SensorContextUtils.java

```diff
@@ -67,15 +65,21 @@ public static Optional<TextRange> toTextRange(InputFile inputFile, SonarAnalyzer
     // https://github.com/dotnet/roslyn/issues/69248
     // https://github.com/dotnet/razor/issues/9051
     // https://github.com/dotnet/razor/issues/9050
-    int lineLength = inputFile.selectLine(startLine).end().lineOffset();
-
-    if (startLineOffset >= lineLength){
+    int startLineLength = inputFile.selectLine(startLine).end().lineOffset();
+    if (startLineOffset > startLineLength){
```

Comment:
If the lines are different we can compute the range instead of dismissing it.

Reply:
I have reworked the logic:
- if the range starts beyond EOL, the start is moved forward, to match the first char of next line
- if the range ends beyond EOL, the end is moved backward, to match the last char of the current line
- a final check is made to make sure that the resulting range is not invalid, returning an empty optional if it is

This way the algorithm is more symmetrical and should be robust (never leading to `InvalidArgumentException` due to range invalidity).